nebula3/gclient/net/Connection.py

```diff
@@ -40,30 +40,34 @@ def __init__(self):
         self._port = None
         self._timeout = 0
         self._ssl_conf = None
+        self.handshakeKey = None
 
-    def open(self, ip, port, timeout):
+    def open(self, ip, port, timeout, handshakeKey=None):
         """open the connection
 
         :param ip: the server ip
         :param port: the server port
         :param timeout: the timeout for connect and execute
+        :param handshakeKey: the server version
         :return: void
         """
-        self.open_SSL(ip, port, timeout, None)
+        self.open_SSL(ip, port, timeout, handshakeKey, None)
 
-    def open_SSL(self, ip, port, timeout, ssl_config=None):
+    def open_SSL(self, ip, port, timeout, handshakeKey=None, ssl_config=None):
         """open the SSL connection
 
         :param ip: the server ip
         :param port: the server port
         :param timeout: the timeout for connect and execute
+        :param handshakeKey: the server version
```

Comment:
handshakeKey: the key for client and server handshake, make sure the handshakeKey is in the white list config of server: client_white_list